internal/util/fieldSetter.go

```diff
@@ -12,6 +12,7 @@ import (
 // Iterates over all parameters and tries to set them
 func DecorateType(serviceType interface{}, fieldNames [][]string) (interface{}, error) {
 	for _, v := range fieldNames {
+		// TODO: this function "re-parses" arguments again
```

Comment:
Not sure what is meant by re-parsing.

Reply:
Splitting the slice into the field path and value